xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp

```diff
@@ -2234,16 +2234,11 @@ bool CActiveAE::RunStages()
             while(!m_vizBuffers->m_outputSamples.empty())
             {
               CSampleBuffer *buf = m_vizBuffers->m_outputSamples.front();
-              if ((now - buf->timestamp) < 0)
```

Comment:
Take care that the vizBuffers don't fill up. This is especially here to drop those that are already way, way behind? E.g. drop those that are > 0.5 seconds behind? Also keep in mind that user can see the delay in the waveform now. Else we could also just add random samples ;-)